engine/src/test/java/org/camunda/bpm/engine/test/bpmn/event/timer/StartTimerEventTest.java

```diff
@@ -1642,6 +1658,110 @@ public void testNonInterruptingFailingTimeCycleInEventSubProcess() {
     assertEquals(2, managementService.createJobQuery().withRetriesLeft().count());
   }
 
+  @Test
+  public void shouldReevaluateTimerCycleWhenDue() throws Exception {
+    // given
+    MyBean myBean = new MyBean("R2/PT1H");
+    processEngineConfiguration.getBeans().put("myBean", myBean);
+    processEngineConfiguration.setReevaluateTimeCycleWhenDue(true);
+
+    createAndDeployProcessWithStartTimer();
+
+    moveByHours(1); // execute first job
+
+    // when bean changed and job is due
+    myBean.setCycle("R2/PT2H");
+    moveByHours(1); // execute second job
+
+    // then one more job is left due in 2 hours
+    Date duedate = managementService.createJobQuery().singleResult().getDuedate();
+    assertThat(duedate)
+        .isEqualToIgnoringMinutes(new Date(ClockUtil.getCurrentTime().getTime() + ((2 * 60 * 1000 * 60))));
```

Comment:
🔧 We could think about adding something to the `ClockUtil` so we can use temporal values instead of repeating those funky calculations for hours manually. One idea would be to add something like `ClockUtil#getCurrentTimePlus` that takes a `TemporalAmount` parameter. Under the hood, it would do something like
```java
public Date getCurrentTimePlus(TemporalAmount amount) {
  return Date.from(getCurrentTime().toInstant().plus(amount));
}
```

You could then write something like this:
```suggestion
        .isEqualToIgnoringMinutes(ClockUtil.getCurrentTimePlus(Duration.ofHours(2l)));
```

Reply:
I like the idea. However, for this ticket, I will only simplify the calculation by using the `TimeUnit.HOURS.toMillis()` methods.
(I just want to avoid spending more time on this and adding test coverage for a new `ClockUtil` method.)